src/Commands/ToggleDone.ts

```diff
@@ -91,7 +92,10 @@ export const toggleLine = (line: string, path: string): EditorInsertion => {
             return { text: line.replace(TaskRegularExpressions.taskRegex, `$1- [${newStatusString}] $4`) };
         } else if (TaskRegularExpressions.listItemRegex.test(line)) {
             // Convert the list item to a checklist item.
-            const text = line.replace(TaskRegularExpressions.listItemRegex, '$1$2 [ ]');
+            const globalFilter = GlobalFilter.get();
+            const addGlobalFilter = globalFilter != '' && !GlobalFilter.includedIn(line);
+            const newTaskText = addGlobalFilter ? `[ ] ${globalFilter}` : '[ ]';
+            const text = line.replace(TaskRegularExpressions.listItemRegex, `$1$2 ${newTaskText}`);
```

Comment:
It just dawned on me that someone recently spent a lot of time moving all code related to the global filter, including manipulating descriptions, in the new `GlobalFilter` class.

If the above cannot be done with the existing methods in GlobalFilter, I feel that the new code should go in to a new method in that class. The existing names there may give ideas.

Reply:
Ah that makes sense. I took a stab at moving as much of the GlobalFilter-related logic to the GlobalFilter class as I could 750df8d9070a0a60b1af589b90f60e49e111bdfb. I didn't go all the way because I wanted to keep the regex-related logic in `ToggleDone.ts`. Let me know if this matches what you're thinking! If not then I'm open to suggestions.